bindings/matrix-sdk-ffi/src/event.rs

```diff
@@ -0,0 +1,195 @@
+use anyhow::{bail, Context};
+use ruma::events::{
+    AnySyncMessageLikeEvent, AnySyncStateEvent, AnySyncTimelineEvent,
+    MessageLikeEventContent as RumaMessageLikeEventContent, RedactContent,
+    RedactedStateEventContent, StaticStateEventContent, SyncMessageLikeEvent, SyncStateEvent,
+};
+
+use crate::{ClientError, MembershipState, MessageType};
+
+pub struct TimelineEvent(pub(crate) AnySyncTimelineEvent);
+
+#[uniffi::export]
+impl TimelineEvent {
+    pub fn event_id(&self) -> String {
+        self.0.event_id().to_string()
+    }
+
+    pub fn sender_id(&self) -> String {
+        self.0.sender().to_string()
+    }
+
+    pub fn event_type(&self) -> Result<TimelineEventType, ClientError> {
+        let event_type = match &self.0 {
+            AnySyncTimelineEvent::MessageLike(event) => {
+                TimelineEventType::MessageLike { content: event.clone().try_into()? }
+            }
+            AnySyncTimelineEvent::State(event) => {
+                TimelineEventType::State { content: event.clone().try_into()? }
+            }
+        };
+        Ok(event_type)
+    }
+}
+
+#[derive(uniffi::Enum)]
+pub enum TimelineEventType {
+    MessageLike { content: MessageLikeEventContent },
+    // we don't support state events yet so they do not have any content
+    State { content: StateEventContent },
```

Comment:
But it says `content: StateEventContent` right there :D

Reply:
let's remove this old comment